File: src/main/java/com/ascargon/rocketshow/api/State.java
package com.ascargon.rocketshow.api;

import com.ascargon.rocketshow.play.CompositionPlayer;
import jakarta.xml.bind.annotation.XmlRootElement;
import lombok.Getter;
import lombok.Setter;

@XmlRootElement
@Getter
@Setter
public class State {

    private Integer currentCompositionIndex;
    private CompositionPlayer.PlayState playState;
    private String currentCompositionName;
    private Long currentCompositionDurationMillis;
    private Long positionMillis;
    private String currentSetName;

    // True while this device is following an incoming MIDI timecode, and the absolute position of
    // that timecode master
    private boolean midiTimecodeLocked;
    private Long midiTimecodeMillis;

    private String error;

}

File: src/main/webapp/src/app/models/state.ts
export class State {
    playState: string = "STOPPED";
    currentCompositionIndex: number = 0;
    currentCompositionName: string = "";
    currentCompositionDurationMillis: number = 0;
    positionMillis: number;
    currentSetName: string;
    midiTimecodeLocked: boolean = false;
    midiTimecodeMillis: number;
    error: string;

    constructor(data?: any) {
        if (!data) {
            return;
        }

        this.playState = data.playState;
        this.currentCompositionIndex = data.currentCompositionIndex;
        this.currentCompositionName = data.currentCompositionName;
        this.currentCompositionDurationMillis = data.currentCompositionDurationMillis;
        this.positionMillis = data.positionMillis;
        this.currentSetName = data.currentSetName;
        this.midiTimecodeLocked = data.midiTimecodeLocked;
        this.midiTimecodeMillis = data.midiTimecodeMillis;
        this.error = data.error;
    }
}
